### PR TITLE
[Workflow management UI changes 2] Workflow pages formset

### PR DESCRIPTION
**Pull request by kaedroho**
This pull request replaces the existing "add to page" view with a formset nested on the workflow form.

Separated from: https://github.com/wagtail/wagtail/pull/6157
Based on: https://github.com/wagtail/wagtail/pull/6169

### CODE REVIEW

File: wagtail/admin/static_src/wagtailadmin/scss/layouts/workflow-edit.scss [hunk 1]
```diff
@@ -0,0 +1,31 @@
+@import '../../../../../../client/scss/tools/mixins.general';
+
+.listing {
+    .field label {
+        @include visuallyhidden;
+    }
+
+    input,
+    select,
+
```
Comment (zerolab) [inline wagtail/admin/static_src/wagtailadmin/scss/layouts/workflow-edit.scss:10]: nitpick: extra blank line

File: wagtail/admin/templates/wagtailadmin/workflows/includes/workflow_pages_formset.html [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        {% for form in formset.forms %}
+            <tr id="inline_child_{{ form.prefix }}"{% if form.DELETE.value %} style="display: none;"{% endif %}>
+                {% if form.non_field_errors %}
+                    <p class="error-message">
+                        {% for error in form.non_field_errors %}
+                            <span>{{ error|escape }}</span>
+                        {% endfor %}
+                    </p>
+                {% endif %}
+                {% include "./workflow_pages_form.html" with form=form only %}
```
Comment (zerolab) [inline wagtail/admin/templates/wagtailadmin/workflows/includes/workflow_pages_formset.html:36]: Let's avoid relative includes everywhere

File: wagtail/admin/templates/wagtailadmin/workflows/task_chooser/includes/results.html [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        <tbody>
+            {% for task in tasks %}
+                <tr>
+                    <td class="title">
+                        <div class="title-wrapper"><a href="{% url 'wagtailadmin_workflows:task_chosen' task.id %}" class="task-choice">{{ task.name }}</a></div>
+                    </td>
+                    {% if searchform.specific_task_model_selected %}
+                        {# TODO #}
+                    {% else %}
+                        <th>
```
Comment (zerolab) [inline wagtail/admin/templates/wagtailadmin/workflows/task_chooser/includes/results.html:39]: Note: this should go away when #6169 is merged

Comment (kaedroho) [reply wagtail/admin/templates/wagtailadmin/workflows/task_chooser/includes/results.html:39]: Yep, I think this PR is based on the initial commit and doesn't include the fixup.

File: wagtail/admin/static_src/wagtailadmin/scss/layouts/workflow-edit.scss [hunk 4]
```diff
@@ -25,3 +25,7 @@
         width: 100%;  // so that 'choose another page' button displays in its entirety
     }
 }
+
+.top-padding {
+    padding-top: 52px;
```
[the remark this answers is not in the document]
Comment (kaedroho) [reply wagtail/admin/static_src/wagtailadmin/scss/layouts/workflow-edit.scss:30]: I think this is for the side padding, but this padding is usually added directly on to the `<fieldset>` tag https://github.com/wagtail/wagtail/blob/master/client/scss/elements/_forms.scss#L19

[the remark this answers is not in the document]
Comment (kaedroho) [reply wagtail/admin/static_src/wagtailadmin/scss/layouts/workflow-edit.scss:30]: I've updated it to follow: https://github.com/wagtail/wagtail/blob/f1dadba75afc0eba41380db5b71080d5881522f6/wagtail/admin/static_src/wagtailadmin/scss/layouts/page-editor.scss#L62

File: wagtail/admin/static_src/wagtailadmin/scss/layouts/workflow-edit.scss [hunk 5]
```diff
@@ -0,0 +1,31 @@
+@import '../../../../../../client/scss/tools/mixins.general';
+@import 'page-editor.scss';
```
Comment (zerolab) [inline wagtail/admin/static_src/wagtailadmin/scss/layouts/workflow-edit.scss:2]: stylelint fails with:

```
wagtail/admin/static_src/wagtailadmin/scss/layouts/workflow-edit.scss
  2:22  ✖  Unexpected extension       scss/at-import-partial-extension-blacklist
                 ".scss" in imported partial name
 ```

so drop `.scss`? also, would it not make sense to include `page-editor.scss` directly in the template? that way it is more explicit

Comment (kaedroho) [reply wagtail/admin/static_src/wagtailadmin/scss/layouts/workflow-edit.scss]: Should already be on the template through `{% include "wagtailadmin/pages/_editor_css.html" %}`

This import is required in order to get the `$object-title-height` variable.

Comment (zerolab) [reply wagtail/admin/static_src/wagtailadmin/scss/layouts/workflow-edit.scss]: hmm, this includes so much extra just for one var. why not move that to `_variables.scss`?

Comment (kaedroho) [reply wagtail/admin/static_src/wagtailadmin/scss/layouts/workflow-edit.scss]: Done! Wonder why it wasn't there in the first place

**Pull request merged**